yield_plugin/src/yield_plugin.cpp

```diff
@@ -0,0 +1,252 @@
+
+      double goal_velocity = rwol_collision[0].object.velocity.twist.linear.x;
+      
+      if (goal_velocity <= config_.min_obstacle_speed){
+        ROS_WARN_STREAM("The obstacle is not moving");
+      }
+
+      double goal_pos = x_lead - goal_velocity * gap_time; 
+
+      double initial_time = 0;
+      double initial_pos = 0.0; //relative initial position (first trajectory point)
+
+      double initial_accel = 0;
+      double goal_accel = 0;
+
+      double delta_v_max = abs(rwol_collision[0].object.velocity.twist.linear.x - max_trajectory_speed(original_tp.trajectory_points));
+      // reference time, is the maximum time available to perform object avoidance (length of a trajectory)
+      double t_ref = (original_tp.trajectory_points[original_tp.trajectory_points.size() - 1].target_time.toSec() - original_tp.trajectory_points[0].target_time.toSec());
+      // time required for comfortable deceleration
+      double t_ph = config_.acceleration_adjustment_factor * delta_v_max / config_.maximum_deceleration_value;
+
+      // planning time for object avoidance
+      double tp = 0;
+
+      if(t_ph > config_.tpmin && t_ref > t_ph)
+      {
+        tp = t_ph;
+      }
+      else if(t_ph < config_.tpmin)
+      {
+        tp = config_.tpmin;
+      }
+      else
+      {
+        tp = t_ref;
+      }
+      
+      ROS_DEBUG_STREAM("Object avoidance planning time: " << tp);
+
+      std::vector<double> values = quintic_coefficient_calculator::quintic_coefficient_calculator(initial_pos, 
+                                                                                                  goal_pos,
+                                                                                                  current_speed_, 
+                                                                                                  goal_velocity, 
+                                                                                                  initial_accel, 
+                                                                                                  goal_accel, 
+                                                                                                  initial_time, 
+                                                                                                  tp);
+
+      std::vector<cav_msgs::TrajectoryPlanPoint> new_trajectory_points;
+      new_trajectory_points.push_back(original_tp.trajectory_points[0]);
+      std::vector<double> original_traj_downtracks = get_relative_downtracks(original_tp);
+      
+      for(size_t i = 1; i < original_tp.trajectory_points.size(); i++ )
+      {
+        double traj_target_time = i * tp / original_tp.trajectory_points.size();
+        double dt_dist = polynomial_calc(values, traj_target_time);
+        double dv = polynomial_calc_d(values, traj_target_time);
+        cav_msgs::TrajectoryPlanPoint new_tpp;
+        // the last moving trajectory point
+        if (dv >= 1.0)
```

Comment:
Should this 1.0 be the config min_speed above? Or some other epsilon constant?

Reply:
They are slightly different. The one above is when we consider the object stationary, and this is when we consider the ego vehicle stopped. I renamed the parameters and added comments for more clarification.